Automattic/wp-calypso
# Checkout: Show G Suite discount callout even if coupon is applied

#### Changes proposed in this Pull Request

The "Discount for first year" text shown in new checkout disappears if a coupon is added to the cart. This fixes that behavior.

The logic previously was: if the price is discounted, and no coupon is applied, display the callout. This was intentional because there is no easy way to determine if the displayed discount is the result of a coupon or a first-year discount (which is a _sale_ coupon).

This PR changes the logic to rely instead of the presence of `is_sale_coupon_applied`, which doesn't guarantee that the sale coupon is for a first-year discount, but it's pretty likely that's the case and will allow us to display the callout even when a coupon is added to the cart.

Before:

![before-discount](https://user-images.githubusercontent.com/2036909/85180244-f1147900-b250-11ea-8484-31014d781cb8.png)

After:

![after-discount](https://user-images.githubusercontent.com/2036909/85180253-f7a2f080-b250-11ea-902e-c1a0d39d9ea3.png)


#### Testing instructions

- Visit checkout with GSuite in your cart.
- Verify that you see "Discount for first year".
- Add a coupon to your cart.
- Verify that you still see "Discount for first year".
